src/main/kotlin/org/objectionary/ddr/sources/Sources.kt

```diff
@@ -0,0 +1,113 @@
+package org.objectionary.ddr.sources
+
+import org.objectionary.ddr.transform.XslTransformer
+import org.slf4j.LoggerFactory
+import org.w3c.dom.Document
+import java.io.File
+import java.io.FileInputStream
+import java.nio.file.Files
+import java.nio.file.Path
+import javax.xml.parsers.DocumentBuilderFactory
+
+private val sep = File.separatorChar
+
+/**
+ * Stores the source files in [Document] format and makes some changes to them
+ */
+interface Sources {
+    /** @property path path to directory with source files or single source file */
+    val path: Path
+
+    /** @property documents all documents */
+    val documents: MutableMap<Document, String>
+
+    /**
+     * Walks through [path] and collect [documents]
+     */
+    fun walkSources()
+}
+
+/**
+ * Stores the source xmir files in [Document] format and makes some changes to them
+ *
+ * @param strPath string path to directory with source files or single source file
+ * @param postfix postfix of the resulting directory
+ * @param gather if outputs should be gathered
+ */
+class SourcesDdr(
+    strPath: String,
+    private val postfix: String = "ddr",
+    private val gather: Boolean = true
+) : Sources {
+    private val logger = LoggerFactory.getLogger(this.javaClass.name)
+
+    /** @property path path to directory with source files or single source file */
+    override val path: Path = Path.of(strPath)
+
+    /** @property documents all documents */
+    override val documents: MutableMap<Document, String> = mutableMapOf()
+
+    /**
+     * Walks through [path], make some xml transformation on xmir files and collect [documents]
+     */
+    override fun walkSources() {
+        val transformer = XslTransformer()
```

Comment:
@c71n93 This is hardcode. Can we make the transformer injectable during object construction?